tests/e2e/run_e2e.sh

```diff
@@ -1,15 +1,24 @@
 #!/bin/bash
-set -eu
+set -euo pipefail
 
-# compile test
-go test -c ./tests/e2e -o ./tests/e2e/e2e.test
-go build -v -o ./tests/e2e/vpp-agent ./cmd/vpp-agent
+echo "preparing E2E test"
+set -x
 
-# start vpp image
+# compile vpp-agent
+go build -v -o ./tests/e2e/vpp-agent.test \
+  -ldflags "-X github.com/ligato/vpp-agent/vendor/github.com/ligato/cn-infra/agent.BuildVersion=TEST_E2E" \
+  ./cmd/vpp-agent
+
+# compile e2e test suite
+go test -c -o ./tests/e2e/e2e.test ./tests/e2e
+
+# start image
+# TODO: do not run docker image with pid=host,
+#  because any other vpp running now breaks tests
```

Comment:
ACK, I'm going to try to get rid of --pid="host" and configure a common process namespace between the agent and microservices. Difficulty is that the PID of a microservice as given by docker daemon will not be the same as seen from the agent point of view, but it should be possible to find a workaround...

Reply:
Perhaps using `--pid=container:$cid` should help here.

```
--pid=""  : Set the PID (Process) Namespace mode for the container,
             'container:<name|id>': joins another container's PID namespace
             'host': use the host's PID namespace inside the container
```
> https://docs.docker.com/engine/reference/run/#pid-settings---pid

Reply:
Yeah, you will need to start microservices with `Pid: "container:<test-container-name>"`

Reply:
fyi: this is the issue that we are facing with non-host pid namespaces: https://stackoverflow.com/questions/52021956/get-internal-docker-pid-by-system-pid

So for example, docker tells us that microservice "my-ms" has PID 5000, but inside the process namespace of the agent's container (that is shared with ms) the PID is something like 53. The mapping can be read from /proc, but mounting /proc from host is effectively the same as using --pid=host.

I'll dig into it more though, since avoiding --pid=host could give us more isolated environments...

Reply:
Oh, just realized that this can't be easily done with current implementation of microservice namespace reference. I opened an issue to continue this discussion: #1485